.github/workflows/test_01.yml

```diff
@@ -3,15 +3,40 @@ name: test01_workflow
 on: push
 
 jobs:
-  build:
+  build_and_output_tests:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - name: Build
         run: g++ -std=c++17 *.cpp
       - name: Make test_01.sh executable
-        run: chmod +x ./test_01.sh
-      - name: Run test
+        run: chmod +x ./test_01.sh      
+      - name: Run test_01.sh
         run: ./test_01.sh
       - name: Diff
-        run: diff output.txt expected_output.txt
\ No newline at end of file
+        run: diff output.txt expected_output.txt
+        
+
+  code_analysis:     
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build
+        run: g++ -std=c++17 *.cpp
```

Comment:
Azért a fordítói warningok is legyenek bekapcsolva, legalább a `-Wall` és a `-Werror` is, hogy ne fogadja el, ha warningot dob a fordító.